library/pk_wrap.c

```diff
@@ -1147,7 +1147,7 @@ static int eckey_check_pair_psa(const mbedtls_ecp_keypair *pub,
                                                       &prv_key_len));
     status = psa_destroy_key(key_id);
     if (ret != 0 || status != PSA_SUCCESS) {
-        return (ret != 0) ? ret : PSA_PK_TO_MBEDTLS_ERR(status);
+        return (status != PSA_SUCCESS) ? PSA_PK_TO_MBEDTLS_ERR(status) : ret;
```

Comment:
Really? I think if both `export_public_key()` and `destroy()` failed, I think we want to return the error from `export_public_key()`.

Reply:
Also, I think I'd find the following more readable:
```c
if (ret != 0) {
    return ret;
} else if (status != PSA_SUCCESS) {
    return PSA_PK_TO_MBEDTLS_ERR(status);
}
```
but perhaps it's a matter of taste.

While at is, I'm not sure using `ret` for the "real"/first error and `status` for the second one is the most readable. A few places in the code use `destruction_status` for the second one, which at least is explicit.

While talking about style, a few lines above we have `status = psa_import_key(...)` and we do the conversion later (`return PSA_PK_TO_MBEDTLS_ERR(status);`), while here we do all at once: `ret = PSA_PK_TO_MBEDTLS_ERR(psa_export_public_key(...)`. I think the second version is not ideal as it pushes things quite far to the right, leaving little room for arguments, so I prefer the first version. Also, I think it's slightly better to be consistent and use the same style everywhere.

(Wow, it looks like I have a lot of opinions about style once I get started on the topic ^^)

Reply:
Ok, I was probably too sleepy when I did this and I misunderstood the comment from Andrzej: so the original version of `pk_wrap` was fine for the order of return codes; it was the `pkparse` one to be in the wrong order.
Sorry for that; I'll fix/revert this